.github/workflows/backport-workflow.yml

```diff
@@ -29,15 +37,36 @@ jobs:
         if: ${{ steps.check_pr_labels.outputs.result == 'true' }}
         uses: actions/checkout@v4
         with:
+          # ensure the backport target branch is checked out, too
           fetch-depth: 0
+          path: repo
+
+      - name: Checkout Backport tool
+        if: ${{ steps.check_pr_labels.outputs.result == 'true' }}
+        uses: actions/checkout@v4
+        with:
+          repository: hazelcast/backport
+          path: backport
 
       - name: Checkout maintenance branch and cherry-pick
         if: ${{ steps.check_pr_labels.outputs.result == 'true' }}
+        working-directory: repo
         run: |
+          # Git metadata is required but not available out-of-the-box, inherit from the action
           git config user.name "${GITHUB_ACTOR}"
           git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
 
-          git fetch
-          git checkout ${{ inputs.target-branch }}
-          git cherry-pick -x --strategy=recursive -X theirs $GITHUB_SHA
-          git push
+          # Add "upstream" remote as checkout action doesn't include by default
+          git remote add upstream "${{ github.event.repository.clone_url }}"
```

Comment:
> Can you point to any docs about this?
> Also default it `origin` and not `upstream`.

It's not really a `checkout` issue - the `backport` action was designed around forked repos, so expects an `upstream` remote - here we aren't working with forks, so there isn't one - [but it's hardcoded to expect one](https://github.com/hazelcast/backport/blob/a71029f8c06d1a2f3f7b404cdbe3992b52faac3f/backport#L114).

We could change the `backport` script but I think it works.
Thinking again though, this kind of internal-implementation-awareness is probably only sensible in ann action within the same repo.

Reply:
> Thinking again though, this kind of internal-implementation-awareness is probably only sensible in ann action within the same repo.

I think this is also the point i am trying to make albeit I didn't state explicitly. 
I only suggested as now `backport` is being used in docs also so thought an action might serve better rather than using script directly. Anyways, it works so its fine. Not a biggy. thanks